plugin/trino-hive/src/main/java/io/trino/plugin/hive/authentication/AuthenticationModules.java

```diff
@@ -33,8 +34,8 @@ private AuthenticationModules() {}
 
     public static Module noHdfsAuthenticationModule()
     {
-        return binder -> binder
-                .bind(HdfsAuthentication.class)
+        return binder -> newOptionalBinder(binder, HdfsAuthentication.class)
+                .setDefault()
```

Comment:
I don't like optional binding. With optional binding it is hard to track what is the actual implementation. As one can override the binding. Maybe instead we could allow to replace the `HiveAuthenticationModule` with something else.

I need to think about it more.

Reply:
The problem with module replacement is that the path by which `HiveAuthenticationModule` ends up being loaded in SEP is convoluted: `StarburstHivePlugin.getConnectorFactories()` -> `HiveConnectorFactory.create()` (reflection) -> `InternalHiveConnectorFactory.createConnector()` -> (a whole bunch of modules including `HiveAuthenticationModule`). Replacing just one module would require copying the class hierarchy: `StarburstHiveConnectorFactory.create()` -> `InternalStarburstHiveConnectorFactory.createConnector()` -> (the same bunch of modules with one replacement). Without serious refactoring on both sides, this would be a nightmare to keep in sync in case of Trino changes.